sccameraview/src/main/java/com/example/sccameraview/Camera2View.java

```diff
@@ -369,6 +379,70 @@ public int compare(Size lhs, Size rhs) {
         }
     }
 
+    @Override
+    public void takePicture() {
+        imageReader = ImageReader.newInstance(previewSize.getWidth(), previewSize.getHeight(), ImageFormat.JPEG, 1);
```

Comment:
The video and preview sizes are chosen basically by the same algorithm after you look at it several times, with the next feature it is unified and only one method is called. 
Please elaborate on why this should not be used. 
I've updated the code based on the other suggestions